src/Widgets/DisplayWidget.vala

```diff
@@ -87,6 +87,43 @@ public class Network.Widgets.DisplayWidget : Gtk.Box {
                 return true;
             });
             break;
+        case Network.State.CONNECTED_MOBILE_WEAK:
+            image.icon_name = "network-cellular-signal-weak-%ssymbolic".printf (secure? "secure-" : "");
+            break;
+        case Network.State.CONNECTED_MOBILE_OK:
+            image.icon_name = "network-cellular-signal-ok-%ssymbolic".printf (secure? "secure-" : "");
+            break;
+        case Network.State.CONNECTED_MOBILE_GOOD:
+            image.icon_name = "network-cellular-signal-good-%ssymbolic".printf (secure? "secure-" : "");
+            break;
+        case Network.State.CONNECTED_MOBILE_EXCELLENT:
+            image.icon_name = "network-cellular-signal-excellent-%ssymbolic".printf (secure? "secure-" : "");
+            break;
+        case Network.State.CONNECTING_MOBILE:
+            animation_timeout = Timeout.add (300, () => {
+                animation_state = (animation_state + 1) % 4;
+                string strength = "";
+                switch (animation_state) {
+                case 0:
+                    strength = "weak";
+                    break;
+                case 1:
+                    strength = "ok";
+                    break;
+                case 2:
+                    strength = "good";
+                    break;
+                case 3:
+                    strength = "excellent";
+                    break;
+                }
+                image.icon_name = "network-cellular-signal-" + strength + (secure? "secure-" : "")  + "-symbolic";
+                return true;
```

Comment:
Explicitly return `true` here makes the timeout run always, even if the state is not `CONNECTING_MOBILE`. I'm not gonna force you to fix this though as this is also the case in master with WiFi.